src/Bicep.Core.Samples/Files/InvalidModules_LF/main.diagnostics.bicep

```diff
@@ -345,8 +345,8 @@ module moduleLoopForRuntimeCheck3 'modulea.bicep' = [for thing in []: {
 //@[7:33) [BCP035 (Error)] The specified "module" declaration is missing the following required properties: "params". |moduleLoopForRuntimeCheck3|
   name: concat(moduleLoopForRuntimeCheck[1].outputs.stringOutputB, moduleLoopForRuntimeCheck[1].outputs.stringOutputA )
 //@[8:119) [prefer-interpolation (Warning)] Dynamic variable should not use concat - string interpolation should be used.\n[See : https://aka.ms/bicep/linter/prefer-interpolation] |concat(moduleLoopForRuntimeCheck[1].outputs.stringOutputB, moduleLoopForRuntimeCheck[1].outputs.stringOutputA )|
-//@[15:65) [BCP120 (Error)] The property "name" must be evaluable at the start of the deployment, and cannot depend on any values that have not yet been calculated. Accessible properties of moduleLoopForRuntimeCheck are "name". |moduleLoopForRuntimeCheck[1].outputs.stringOutputB|
-//@[67:117) [BCP120 (Error)] The property "name" must be evaluable at the start of the deployment, and cannot depend on any values that have not yet been calculated. Accessible properties of moduleLoopForRuntimeCheck are "name". |moduleLoopForRuntimeCheck[1].outputs.stringOutputA|
+//@[15:51) [BCP120 (Error)] The property "name" must be evaluable at the start of the deployment, and cannot depend on any values that have not yet been calculated. Accessible properties of moduleLoopForRuntimeCheck are "name". |moduleLoopForRuntimeCheck[1].outputs|
+//@[67:103) [BCP120 (Error)] The property "name" must be evaluable at the start of the deployment, and cannot depend on any values that have not yet been calculated. Accessible properties of moduleLoopForRuntimeCheck are "name". |moduleLoopForRuntimeCheck[1].outputs|
```

Comment:
I like the change to the span that the squiggly is being displayed on. Seeing it here, I wonder if there's something we can do to make this message a bit more obvious - given that the squiggly is being displayed on the value rather than the key, it's not immediately obvious that `name` in the message relates to the property value assignment, which could be especially hard to follow on a larger expression.

What about something like: 'This expression is being used in an assignment to the "<key>" property of the "<type>" type, which requires a value that can be calculated at the start of the deployment. Accessible properties...<rest of message>'

I'm interested to hear other people's thoughts here as well - worth spending a bit of time getting right I think!

Reply:
I like it.

Reply:
Good suggestion! Created https://github.com/Azure/bicep/issues/2863 to track it. Will submit another PR to implement it.